docs/getting_started/kernel.md

```diff
@@ -0,0 +1,337 @@
+| WithLoggerMetricHandler | Adds a default log metric handler |
+| WithLoggerSentryHandler | Adds a default logger sentry handler |
+| WithMetricDaemon | Adds the "metric" module to this _Kernel_ |
+| WithProducerDaemon | Adds a default producer daemon |
+| WithTracing | Adds a default logger tracingHandler |
+| WithUTCClock | Makes the _Kernel_ use clock.WithUseUTC(useUTC) |
+
+Similarly, package _application_ offers two helper functions for running a module, that internally will make used of `application.Default()`:
+
+[embedmd]:# (../../pkg/application/runners.go /func RunModule/ /\n}/)
+```go
+func RunModule(name string, moduleFactory kernel.ModuleFactory, options ...Option) {
+	app := Default(options...)
+	app.Add(name, moduleFactory)
+	app.Run()
+}
+```
+
+This functions instantiates the `Default()` kernel, adds all options to it, adds the module then runs that kernel.
+
+[embedmd]:# (../../pkg/application/runners.go /func RunApiServer/ /\n}/)
+```go
+func RunApiServer(definer apiserver.Definer, options ...Option) {
+	options = append(options, WithExecBackoffSettings(&exec.BackoffSettings{
+		InitialInterval: time.Millisecond * 100,
+		MaxElapsedTime:  time.Second * 10,
+		MaxInterval:     time.Second,
+	}))
+
+	RunModule("api", apiserver.New(definer), options...)
+}
+```
+
+This function configures some exec.BackoffSettings, then creates a module out of `definer`, and lastly it calls `RunModule` for that module.
+
+### Configuration files
+
+The code bellow passes a config file to the call that creates a _Kernel_:
+
+[embedmd]:# (../../examples/apiserver/simple-handlers/main.go /func main/ /\n}/)
+```go
+func main() {
+	app := application.New(application.WithConfigFile("config.dist.yml", "yml"))
+	app.Add("api", apiserver.New(apiDefiner))
+	app.Run()
+}
+```
+
+This is a convenient way to store configuration in an external configuration file and further separate the code from its configurations. Below is an example of such a .yml config file:
+
+```
+env: dev
+app_project: project
+app_family: family
+app_name: name
+
+api:
+  port: 8088
+  mode: debug
+  timeout:
+    read: 5s
+    write: 5s
+    idle: 5s
+
+api_auth_keys:
+  - changeMe
+api_auth_basic_users:
+  - admin:password
+```
+
+Notice how it has three sections: firstly it defines app relates configuration values, then come the "api" section, and lastly the two auth keys. Intuitively we can see that configurations used in three different sections of our code were merged into the same file, and only those prefixed with `app_` are needed to create the _Kernel_. 
+
+### Stopping a Kernel
+
+The last three methods of the _Kernel_ interface deserve attention as well:
+
+- `Running() <-chan struct{}`
+  * `Running` is used to tell the state of a _Kernel_. This method returns an unbuffered channel that blocks until the _Kernel_ has started, because the `Run` method closes it the moment the _Kernel_ is up and running. 
```

Comment:
What does up and running mean? It means that all modules have been spawned, but they might not yet have run a single instruction (depending on the go scheduler)